src/config.ts

```diff
@@ -1,10 +1,11 @@
 const hasDOM = typeof document !== 'undefined' && typeof window !== 'undefined';
-const scriptElement: HTMLScriptElement = hasDOM && document.currentScript;
+const scriptElement: HTMLOrSVGScriptElement | null = document?.currentScript;
 // @ts-ignore
-const globalConfig: Record<string, PrismConfig[keyof PrismConfig]> =
-	globalThis.Prism?.constructor?.name === 'Object' ? globalThis.Prism : {};
+const globalConfig: Record<string, PrismConfig[keyof PrismConfig] | null> =
+	globalThis.Prism?.constructor?.name === 'Object' ? globalThis.Prism.config : {};
 
 function getGlobalSetting (name: string) {
+	// eslint-disable-next-line regexp/no-unused-capturing-group
```

Comment:
ESLint still can't analyze the code properly. 😅

Reply:
Why haven't we disabled this rule?